common4j/src/main/com/microsoft/identity/common/java/controllers/ExceptionAdapter.java

```diff
@@ -436,6 +439,27 @@ public static ClientException clientExceptionFromException(@NonNull final Throwa
 
 
         if (e instanceof OutOfMemoryError) {
+            Logger.warn(TAG,
+                    "Received an out of memory error, attempting to attach stacktrace...");
+
+            // Record exception stacktrace in span
+            // Put this in a try-catch in case this telemetry emit attempt causes another OOM
+            try {
+                final Span currentSpan = SpanExtension.current();
+
+                // Attach the stack trace, to debug in telemetry later
+                currentSpan.setAttribute(
+                        AttributeName.out_of_memory_exception_stacktrace.name(),
+                        StringUtil.getStacktraceAsStringFromElementArray(e.getStackTrace())
+                );
+
+                Logger.warn(TAG,
+                        "Received an out of memory error, stacktrace attached to span with id: " + currentSpan.getSpanContext().getSpanId());
+            } catch (Throwable throwable) {
+                Logger.error(TAG,
+                        "Failed to emit telemetry for out of memory exception.", throwable);
```

Comment:
maybe Logger.warn or Logger.error without throwable? To notmake Logger heavy on memory